lib/apiv2/node.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiv2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/projectcalico/libcalico-go/lib/numorstring"
+)
+
+const (
+	KindNode     = "Node"
+	KindNodeList = "NodeList"
+)
+
+// Node contains information about a Node resource.
+type Node struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	Metadata metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of the Node.
+	Spec NodeSpec `json:"spec,omitempty"`
+}
+
+// NodeSpec contains the specification for a Node resource.
+type NodeSpec struct {
+	// BGP configuration for this node.  If this omitted, the Calico node
+	// will be run in policy-only mode.
+	BGP *NodeBGPSpec `json:"bgp,omitempty" validate:"omitempty"`
+}
+
+// NodeSpec contains the specification for the Node BGP configuration.
+type NodeBGPSpec struct {
+	// The AS Number of the node.  If this is not specified, the global
+	// default value will be used.
+	ASNumber *numorstring.ASNumber `json:"asNumber,omitempty"`
```

Comment:
Are we sticking with our custom `numorstring` package or should we be switching to just use the upstream `intstr` package from k8s?